Extra/Shaders/NewVegasReloaded/Effects/Includes/aioNAL_ColorFilter.hlsl

```diff
@@ -0,0 +1,82 @@
+#include "Includes/aioNAL_Utilities.hlsl"
+
+//ColorFilterValues.x=UseColorSaturation, y=HueMid, z=HueRange, w=SaturationLimit
+
+void prod80ColorFilter(inout float3 color, float3 colorMod, float4 ColorFilterValues, float ColorFilterStrength)
+{
+
+	 colorMod = saturate( color.xyz );
+	 float greyVal = grayValue( colorMod.xyz );
+	 float colorHue = Hue( colorMod.xyz );
+	 
+	 float colorSat = 0.0f;
+	 float minColor = min( min ( colorMod.x, colorMod.y ), colorMod.z );
+	 float maxColor = max( max ( colorMod.x, colorMod.y ), colorMod.z );
+	 maxColor = max( max ( colorMod.x, colorMod.y ), colorMod.z );
+	 float colorDelta = maxColor - minColor;
+	 float colorInt = ( maxColor + minColor ) * 0.5f;
+	 
+	 if ( colorDelta != 0.0f )
+	 {
+			if ( colorInt < 0.5f )
+				 colorSat = colorDelta / ( maxColor + minColor );
+			else
+				 colorSat = colorDelta / ( 2.0f - maxColor - minColor );
+	 }
+	 
+	 //When color intensity not based on original saturation level
+	 colorSat = lerp(1.0f, colorSat, ColorFilterValues.x);
+	 
+	 float4 hueMinMax = float4(0.0f, 0.0f, 0.0f, 0.04f);//x = hueMin_1, y = hueMin_2, z = hueMax_1, w = hueMax_2;
+	 
+	 if ( ColorFilterValues.z > ColorFilterValues.y )
```

Comment:
are we branching based on constant params across the entire frame? Because if we define them as uniform, I think the branching has no impact